From 42928117d27372abc661c7f4cbb3b4f9528423ff Mon Sep 17 00:00:00 2001
From: Sven Joachim <svenjoac@gmx.de>
Date: Mon, 9 Jan 2023 20:14:34 +0100
Subject: [PATCH] Look in debian/tmp for files to move

Commit ef2b851418d0 introduced a new feature where filenames prefixed
with => would be moved rather than copied from the source to the
destination.  Unfortunately it would fail when the files are located
in debian/tmp, which is the most common use case for this feature.

Also add a new test to ensure that this feature works.

Closes: #1028347
---
 lib/dh-exec-install-rename |  2 ++
 t/test_install_move.bats   | 13 +++++++++++++
 2 files changed, 15 insertions(+)

diff --git a/lib/dh-exec-install-rename b/lib/dh-exec-install-rename
index 7d4d993..d8b0355 100755
--- a/lib/dh-exec-install-rename
+++ b/lib/dh-exec-install-rename
@@ -63,6 +63,8 @@ if (/([^\s]*)\s+=>\s+([^\s]*)/ || /^=>\s+([^\s]*)/) {

         if ($move_this) {
             move ($src, File::Spec->catfile ($debpath, $dstfile)) or
+		move (File::Spec->catfile ("debian/tmp", $src),
+		      File::Spec->catfile ($debpath, $dstfile)) or
                 die "Failed to move '$src': $!";
         } else {
             cp ($src, File::Spec->catfile ($debpath, $dstfile)) or
diff --git a/t/test_install_move.bats b/t/test_install_move.bats
index 93b66e7..298340d 100755
--- a/t/test_install_move.bats
+++ b/t/test_install_move.bats
@@ -25,6 +25,19 @@ EOF
     ! [ -f var/lib/dh-exec/test-output ]
 }

+@test "install: moving from debian/tmp works" {
+    install -d debian/tmp/var/lib/dh-exec
+    echo foo >debian/tmp/var/lib/dh-exec/test-output
+
+    run_dh_exec_with_input .install <<EOF
+#! ${top_builddir}/src/dh-exec-install
+=> var/lib/dh-exec/test-output
+EOF
+
+    expect_file "/var/lib/dh-exec/test-output"
+    ! [ -f debian/tmp/var/lib/dh-exec/test-output ]
+}
+
 @test "install: rename takes priority over move" {
     install -d var/lib/dh-exec
     echo foo >var/lib/dh-exec/test-output
--
2.39.0

